### ISSUE TITLE
GenerateSourcesTask.runtimeJar is annotated with InputFile even though it's an output

### ISSUE THREAD

**Issue by Juuxel** [FabricMC/fabric-loom#821]
https://github.com/FabricMC/fabric-loom/blob/0fa4e1e05ed860bcffb0ec146e8c8d9121006cc8/src/main/java/net/fabricmc/loom/task/GenerateSourcesTask.java#L99-L100

It's mostly written to, and I don't think Gradle tasks are expected to modify their *inputs*. The only reading operation on it writes to it right afterwards:

https://github.com/FabricMC/fabric-loom/blob/0fa4e1e05ed860bcffb0ec146e8c8d9121006cc8/src/main/java/net/fabricmc/loom/task/GenerateSourcesTask.java#L285-L289

**Comment by modmuss50** [comment]
Yeah, this isnt ideal tasks should really not have the same input and output.

I think I want to change genSources to keep a copy of the old input jar around before applying the line map. Im not quite sure of the best way to go about doing this 🤔 

I think this will be something to explore for a 1.2, as I dont think its causing a major issue atm?

**Comment by modmuss50** [comment]
runtime jar is no longer a thing, this should be all good now.
